os_migrate/plugins/module_utils/workload_common.py

```diff
@@ -28,44 +28,12 @@
     DEVNULL = open(os.devnull, 'r+', encoding='utf8')
 
 
-def check_process(pid):
-    """Check whether pid exists in the current process table."""
-    if pid is None:
-        return False
-    try:
-        os.kill(int(pid), 0)
-    except (OSError, ValueError):
-        return False
-    else:
-        return True
-
-
-def check_and_cleanup_lockfiles():
-    """
-    Removes lockfiles found in specific directories.
-    If a lockfile is not being used by a process,
-    it is either deleted or not being used.
-    """
-    # Remove specific lockfiles
-    for lockfile in [ATTACH_LOCK_FILE_SOURCE, ATTACH_LOCK_FILE_DESTINATION, PORT_LOCK_FILE]:
-        try:
-            with open(f"{lockfile}.lock", "r", encoding='utf-8') as f:
-                pid = f.read().strip()
-            if check_process(pid):
-                # print(f"Lockfile {lockfile} is currently in use by process: {pid}")
-                pass
-            else:
-                os.remove(lockfile + '.lock')
-                return True
-        except FileNotFoundError:
-            return False
-
-
 def use_lock(lock_file):
     """ Boilerplate for functions that need to take a lock. """
     def _decorate_lock(function):
         def wait_for_lock(self):
-            if check_and_cleanup_lockfiles():
+            time.sleep(1)
```

Comment:
Putting time.sleep(1) here will make the function always block for one second at least, but it will not fix the problem. The problem is with the `for second in range(DEFAULT_TIMEOUT):` below, one of the possible fixes should be what i suggested in #611 .